StrategyGame/Model/Heroes/IHero.cs

```diff
@@ -1,15 +1,17 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.ComponentModel;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace StrategyGame.Model.Heroes
 {
-    public interface IHero
+    public interface IHero : INotifyPropertyChanged
```

Comment:
Why do you have IHero extending interface INotifyPropertyChanged? I think it would be better to create some abstract class and implement both IHero and INotifyPropertyChanged and also implement some basic functionality inside of it.